app/auth/resetPassword.js

```diff
@@ -34,35 +35,14 @@ const ResetPassword = () => {
     const [showPassword, setShowPassword] = useState(false)
 
     useEffect(() => {
-        if (passwordReset) {
-            router.replace('/auth/login')
-        }
-    }, [passwordReset])
-
-    useEffect(() => {
-        if (authError) {
-            setErrorText(authError)
+        if (authError.type === 'resetPassword') {
+            setErrorText(authError.message)
         }
     }, [authError])
 
     useEffect(() => {
         if (password.length) {
-            const requirements = []
-            if (password.length > 0 && password.length < 8) {
-                requirements.push('At least 8 characters')
-            }
-            if (!password.match(/[0-9]/)) {
-                requirements.push('One number')
-            }
-            if (!password.match(/[A-Z]/)) {
-                requirements.push('One uppercase letter')
-            }
-            if (!password.match(/[a-z]/)) {
-                requirements.push('One lowercase letter')
-            }
-            if (!password.match(/[!@#$%^&*]/)) {
-                requirements.push('One special character - !@#$%^&*')
-            }
+            const requirements = checkPasswordRequirements(password)
```

Comment:
I don't think it's necessary to call this onSubmitPress, since every time the password gets changed, the useEffect hook will get called, which will update the passwordRequirements array. And that's the array we check when onSubmitPress is called. If you think I'm wrong or just think it safer, I can add an additional checkPasswordRequirements though.

As far as disabling the button goes, I'm wary of doing so. A lot of UX designs these days stray away from disabling buttons, since they're often confusing to the user, as they don't tell the user what is wrong. Read more here: https://www.smashingmagazine.com/2021/08/frustrating-design-patterns-disabled-buttons/

Reply:
So is there anything stopping a user from clicking create account even when they have problems with the password?

Reply:
Yeah, there's a check in the button press handler method. If passwordRequirements has length greater than 0, the method will return early.